File: frontend/src/app/services/route.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {newRouteModel} from "../models/dto/newRoute.model";
import {Observable} from "rxjs";
import {NewOrderModel} from "../models/dto/newOrder.model";
import {AlternativeRouteModel} from "../models/dto/alternativeRoute.model";

@Injectable({
  providedIn: 'root'
})
export class RouteService {

  constructor(private http: HttpClient) { }

  createRoute(route: newRouteModel): Observable<Boolean>{
    return this.http.post<Boolean>("/api/route", route);
  }

  getAlternativeRoutes(newOrder: NewOrderModel): Observable<AlternativeRouteModel[]>{
    return this.http.post<AlternativeRouteModel[]>("/api/order/alternative", newOrder);
  }
}

File: backend/src/main/java/com/project/logistics/service/RouteService.java
package com.project.logistics.service;

import com.project.logistics.dto.AlternativeRoute;
import com.project.logistics.dto.NewOrder;
import com.project.logistics.dto.NewRouteDto;

import java.util.List;

public interface RouteService {
    boolean createRoute(NewRouteDto routeDto);
    List<AlternativeRoute> getAlternativeRoutes(NewOrder newOrder);
}
